SpringProject/src/test/java/com/gradproject2019/AuthUtilsTest.java

```diff
@@ -0,0 +1,59 @@
+package com.gradproject2019;
+
+import com.gradproject2019.users.service.UserServiceImpl;
+import com.gradproject2019.utils.AuthUtils;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class AuthUtilsTest {
+
+    @Test
+    public void shouldCheckPasswordIsValidAgainstRegex() {
+        Assert.assertFalse(AuthUtils.validate("1@Aa", UserServiceImpl.PASSWORD_PATTERN));
+        Assert.assertFalse(AuthUtils.validate("@Password", UserServiceImpl.PASSWORD_PATTERN));
+        Assert.assertFalse(AuthUtils.validate("1@password", UserServiceImpl.PASSWORD_PATTERN));
+        Assert.assertFalse(AuthUtils.validate("1Password", UserServiceImpl.PASSWORD_PATTERN));
+
+        Assert.assertTrue(AuthUtils.validate("1@Password", UserServiceImpl.PASSWORD_PATTERN));
+
+    }
+
+    @Test
+    public void shouldCheckEmailIsValidAgainstRegex () {
+        Assert.assertTrue(AuthUtils.validate("Gracesophia!1@gmail.com", UserServiceImpl.EMAIL_PATTERN));
+
+        Assert.assertFalse(AuthUtils.validate("gracesophiagmail.com", UserServiceImpl.EMAIL_PATTERN));
+        Assert.assertFalse(AuthUtils.validate("gracesophia@gmailcom", UserServiceImpl.EMAIL_PATTERN));
+        Assert.assertFalse(AuthUtils.validate("gracesophia @gmailcom", UserServiceImpl.EMAIL_PATTERN));
+    }
+
+    @Test
+    public void shouldCheckUsernameNoSpaces() {
+        Assert.assertFalse(AuthUtils.validate("Sophia Grace", UserServiceImpl.USERNAME_PATTERN));
+        Assert.assertTrue(AuthUtils.validate("SophiaGrace", UserServiceImpl.USERNAME_PATTERN));
+    }
+
+    @Test
+    public void shouldCheckPasswordMatchesHashedPassword() {
+        //given
+        String password = "password123";
```

Comment:
This could be extracted into it's own final static field

Reply:
Done!